posthog/migrations/0307_add_indirect_person_override_constraints.py

```diff
@@ -0,0 +1,92 @@
+import django.db.models.expressions
+from django.db import migrations, models
+
+# The previous migration attempted to add a constraint to the personoverride
+# table. We want to remove that constraint, as the ForeignKey replaces this.
+# The function was never used, but to make migrations work we need these two
+DROP_FUNCTION_FOR_CONSTRAINT_SQL = "DROP FUNCTION is_override_person_not_used_as_old_person"
+CREATE_FUNCTION_FOR_CONSTRAINT_SQL = f"""
+CREATE OR REPLACE FUNCTION is_override_person_not_used_as_old_person(team_id bigint, override_person_id uuid, old_person_id uuid)
+RETURNS BOOLEAN AS $$
+  SELECT false;
+$$ LANGUAGE SQL;
+"""
+
+DROP_FUNCTION_FOR_CONSTRAINT_SQL = "DROP FUNCTION is_override_person_not_used_as_old_person"
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("posthog", "0306_featureflag_dashboard"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="PersonOverrideMapping",
+            fields=[
+                ("id", models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name="ID")),
+                ("uuid", models.UUIDField()),
+                ("team_id", models.BigIntegerField()),
+            ],
+        ),
+        migrations.RemoveConstraint(
+            model_name="personoverride",
+            name="unique override per old_person_id",
+        ),
+        migrations.RemoveConstraint(
+            model_name="personoverride",
+            name="old_person_id_is_not_override_person_id",
+        ),
+        migrations.RunSQL(DROP_FUNCTION_FOR_CONSTRAINT_SQL, CREATE_FUNCTION_FOR_CONSTRAINT_SQL),
+        migrations.RemoveField(model_name="personoverride", name="old_person_id"),
+        migrations.AddField(
+            model_name="personoverride",
+            name="old_person_id",
+            field=models.ForeignKey(
+                db_column="old_person_id",
+                on_delete=django.db.models.deletion.CASCADE,
+                related_name="person_override_old",
+                to="posthog.personoverridemapping",
+            ),
+        ),
+        migrations.RemoveField(model_name="personoverride", name="override_person_id"),
+        migrations.AddField(
+            model_name="personoverride",
+            name="override_person_id",
+            field=models.ForeignKey(
+                db_column="override_person_id",
+                on_delete=django.db.models.deletion.CASCADE,
+                related_name="person_override_override",
+                to="posthog.personoverridemapping",
+            ),
+        ),
+        migrations.AddConstraint(
+            model_name="personoverride",
+            constraint=models.UniqueConstraint(
+                fields=("team", "old_person_id"), name="unique override per old_person_id"
+            ),
+        ),
+        # Provides operator classes for integers (gist_int4_ops)
+        migrations.RunSQL("CREATE EXTENSION btree_gist", "DROP EXTENSION btree_gist"),
```

Comment:
are these extensions all standard? ie in a standard postgres distribution

Reply:
All good ✅ 

https://www.postgresql.org/docs/current/btree-gist.html

Reply:
They are in the PG distribution since at least version 9.6: https://www.postgresql.org/docs/9.6/contrib.html

Reply:
However, it can depend on our specific deployment:
> If you are using a pre-packaged version of PostgreSQL, these modules are typically made available as a separate subpackage, such as postgresql-contrib.